### ISSUE TITLE
Linux: 'change' event not fired when file is first created

### ISSUE THREAD

**Issue by borisovg** [paulmillr/chokidar#271]
```javascript
var chokidar = require('chokidar');

var file = 'test.txt',
        watcher = chokidar.watch(file);

watcher.on('unlink', function () {
        console.log('deleted');
});

watcher.on('change', function () {
        console.log('changed');
});
```

When program is started `test.txt` does not exist. The first time it is created the 'change' event is not fired. After that deleting and recreating the file fires the `change` event.

```
$ rm -f test.txt && node index.js                                                                                                                         
deleted
changed
```

```
$ touch test.txt                                                                                                                                          
$ rm test.txt                                                                                                                                             
$ touch test.txt
```

Node version is 0.10.38.

**Comment by es128** [comment]
That's what the `add` event is for :)

**Comment by borisovg** [comment]
So why does then change fire the second time file is created?

**Comment by borisovg** [comment]
OK, so the fact that even `change` fires before `add` looks to be a duplicate of #261.

**Comment by es128** [comment]
You weren't even listening for `add` in your example script. `change` fired because the remove and re-add was almost immediate - this is expected behavior.

**Comment by borisovg** [comment]
Erm, I waited 5 seconds between each command - is that your definition of "immediate"? ;-)

**Comment by es128** [comment]
`$ rm -f test.txt && node index.js`

That's immediate

**Comment by borisovg** [comment]
That is also run _before_ the script is even started. The result is _exactly_ the same if you delete the file 5s before starting the script.

**Comment by borisovg** [comment]
Here is a modified test with an `add` watcher added (hehe):

**Comment by borisovg** [comment]
Here's a modified example with the `add` watcher added (hehe):

```javascript
var chokidar = require('chokidar');

var file = 'test.txt',
        watcher = chokidar.watch(file);

watcher.on('unlink', function () {
        console.log('deleted');
});

watcher.on('change', function () {
        console.log('changed');
});

watcher.on('add', function () {
        console.log('created');
});
```

```
$ rm -f test.txt && sleep 5 && node index.js 
created
deleted
changed
created
```

```
$ touch test.txt && sleep 5 && rm test.txt && sleep 5 && touch test.txt
```

**Comment by es128** [comment]
Yes, in this last example it looks like you've added evidence for #261. Thanks, it should help with debugging it.
